basic setup, plan to add functionality soon

TechPrep-Moh77 opened the pull request:
## 📚 CTP Habit Tracker Submission

### 📝 Student Information
- **Name**:  Mohamed Kamagate
- **Cohort**: (e.g., C11, C12) C11
- **GitHub Username**: 
- **Week Submission**: (Week 1 / Week 2 / Week 3 / Final)

### 🎯 Implementation Summary
<!-- Provide a brief overview of what you implemented this week -->


### ✅ Completed Features
<!-- Check off the features you've implemented -->

#### Week 1 Features
- [ ] Basic HTML structure with habit forms
- [ ] Add new habits functionality
- [ ] Mark habits as complete/incomplete
- [ ] Streak calculation (current and longest)
- [ ] Local storage persistence
- [ ] View habit history
- [ ] Delete habits functionality
- [ ] Edit existing habits
- [ ] Basic CSS styling

#### Week 2 Features (if applicable)
- [ ] TypeScript conversion complete
- [ ] Type definitions for all data structures
- [ ] Habit categories implemented
- [ ] Difficulty levels and point system
- [ ] Weekly/monthly goal setting
- [ ] Statistics calculations
- [ ] Data export functionality
- [ ] Enhanced styling

#### Week 3 Features (if applicable)
- [ ] React component architecture
- [ ] Interactive dashboard
- [ ] Data visualization charts
- [ ] Calendar view for habit tracking
- [ ] Modals for habit creation/editing
- [ ] Settings panel
- [ ] Achievement system
- [ ] Responsive design
- [ ] Smooth animations

### 🚀 Additional Features
<!-- List any extra features you implemented beyond the requirements -->


### 🤔 Challenges Faced
<!-- Describe any significant challenges you encountered and how you solved them -->


### 💡 What I Learned
<!-- Share the most important things you learned during this week's implementation -->


### 📸 Screenshots
<!-- Optional: Add screenshots of your application to showcase your work -->
<!-- You can drag and drop images directly into this text area -->


### 🧪 Testing Checklist
- [ ] All CRUD operations work correctly
- [ ] Data persists after page reload
- [ ] Application works on mobile devices
- [ ] No console errors in browser
- [ ] Code follows proper naming conventions

### 📁 File Structure Verification
- [ ] All code is in `/habit-tracker/<cohort>/<github-username>/`
- [ ] No files outside of my designated directory
- [ ] Template files were not modified

### 💭 Questions or Comments
<!-- Any questions for instructors or additional comments about your submission -->


---
**Note**: Make sure your code is committed and pushed to your fork before creating this PR!

TechPrep-Moh77 committed:
basic setup, plan to add functionality soon

TechPrep-Moh77 committed:
initial commit restart

shiftyp reviewed (changes requested):
Good start! Just one change currently

TechPrep-Moh77 committed:
Fixed update(previously didnt stage the right files)

TechPrep-Moh77 commented:
### Most recent update(abb8ad6 on 8/15/25) Please let me know if addb8ad6 went through properly or if its the same files

**What's been added:**

- Added more functionality in general
- Implemented CRUD operations such as adding habit, and removing habit 
- Completed button has functionality; pressing it will update completion count for specific habit
- Tweaked html buttons styling (greenyellow for completion button and organge-ish coloring for add habit button)
- Added placeholder image

**Potential problems:**

- Adding a habit is done by adding a new div block to the html file each time. I worry that this may take too much memory
- Placeholder image isn't a good representation of user's added habit


**Planning to add:**

- More functionality in general
- reimplementing the target streak giving by mentor(ryan) to keep track of target streak
- adding more "juice" (game dev term for satisfying feedback) to the overall layout. ex: alert or popup once user reachers their target streak
- Removing the habit input on the top row and fully committing to the one in the blue-grayish layer
- more local storage implementation in general

Again, please let me know if i commited the right files this time(or any useful suggestions I should remove, add, or tweak), the webpage should look like this 
<img width="1918" height="917" alt="habittrackerUpdate" src="https://github.com/user-attachments/assets/c8c20f76-0670-4a52-af2b-7333540b336a" />

shiftyp commented on habit-tracker/C11/TechPrep-Moh77/main.js:
Might want to fix the formatting in this file, indentation specifically. VSCode and Codespaces by extension have a format utility. Hit (CTRL or Command) + P for the command input, then type in "format".

shiftyp reviewed (changes requested):
It looks like you have some changes still in the template directory. Delete your existing template directory, then do `git checkout HEAD~4 -- habit-tracker/template`. That and the formatting change and it'll be mergable!

shiftyp commented on package-lock.json:
Delete this file

TechPrep-Moh77 commented:
Hello, my changes are in C11/TechPrep-Moh77( my user name), so when I
delete template and do  "git checkout origin/main --
habit-tracker/template" , will it reflect the changed in my C11 file?


On Fri, Aug 22, 2025, 10:11 AM Ryan Kahn ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> On package-lock.json
> <https://github.com/CUNYTechPrep/ctp-wd-summer-prep/pull/6#discussion_r2293849469>
> :
>
> Delete this file
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/CUNYTechPrep/ctp-wd-summer-prep/pull/6#pullrequestreview-3144663699>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BU6UTSMUC7ZEIZ7Y5SSQOCT3O4QHXAVCNFSM6AAAAACDA3WKHCVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZTCNBUGY3DGNRZHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

TechPrep-Moh77 committed:
deleted template and packagelock

TechPrep-Moh77 committed:
deleted template
